apps/publisher/modules/asset-action-api.js

```diff
@@ -251,8 +251,19 @@ var api = {};
         }
         var content = req.getContent();
         validatePayload(content);
+        var data = parse(content);
+        var existingAsset = assetAPI.get(options,req,res,session);
+        var newVersion = data.attributes.overview_version;
+        var resources = am.getAssetGroup(existingAsset);
+        for (var index in resources) {
+            if (resources.hasOwnProperty(index)) {
+                var resource = resources[index];
+                if(resource.version === newVersion){
```

Comment:
Thanks, I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/842/commits/ca69fedaf2e24d1f028cf4b3300e5f66df25ce60)